File: javascript/isogram/isogram.js
var Isogram = function(input) {
    this.phrase = input;
};

Isogram.prototype.isIsogram = function() {
    var lowerPhrase = this.phrase.toLowerCase();
    lowerPhrase = lowerPhrase.replace(/[ ]/g, "");
    lowerPhrase = lowerPhrase.replace(/[-]/g, "");
    var usedCharacters = [];
    for (var i = 0; i < lowerPhrase.length; i++) {
        for (var j = 0; j < usedCharacters.length; j++) {
            if (usedCharacters[j] === lowerPhrase[i]) {
                return false;
            }
        }
        usedCharacters.push(lowerPhrase[i]);
    }
    return true;
}

module.exports = Isogram;

File: python/isogram/isogram.py
import re

def is_isogram(phrase):
    lowerPhrase = phrase.lower();
    lowerPhrase = re.sub(r' ','',lowerPhrase);
    lowerPhrase = re.sub(r'-','',lowerPhrase);
    usedCharacters = []
    for i in range(0,len(lowerPhrase)):
        for j in range (0,len(usedCharacters)):
            if usedCharacters[j] == lowerPhrase[i]:
                return False;
        usedCharacters.append(lowerPhrase[i])
    return True
    pass
